libtransact/src/context/manager.rs

```diff
@@ -222,52 +193,59 @@ impl ContextManager {
     }
 }
 
-pub mod sync {
-    //! This module provides a thread-safe ContextManager.
-    //!
-    //! For many uses of the context manager, it will need to be shared between multiple threads,
-    //! with some threads reading and writing to a context while others create contexts.
+pub mod lifecycle {
```

Comment:
I think we should keep modules in their own files. Which would give us context/manager/mod.rs, context/manager/lifecycle.rs, context/manager/sync.rs, context/manager/thread.rs.

We could move things around in a separate PR after this one.

Reply:
I can separate the sync module into its own file in a separate pr and then move the lifecycle module into another file for this pr.